core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -952,6 +952,12 @@ private[spark] class SparkSubmit extends Logging {
     } catch {
       case t: Throwable =>
         throw findCause(t)
+    } finally {
+      try {
+        org.apache.spark.SparkContext.getOrCreate().stop()
```

Comment:
Just one comment - use `SparkContext.getActive.foreach(_.stop())` instead.
Will let others comment about rest of the PR/jira.

Reply:
+1 for @mridulm comment. The root cause of current UT failures might be due to it.